File: spring/querydsl-demo/src/main/java/playground/querydsl/Artist.java
package playground.querydsl;

import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import java.time.LocalDate;
import java.util.UUID;

@Entity
public class Artist {

    @Id
    private UUID id;
    private String name;
    private LocalDate activeFrom;
    private LocalDate activeTo;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public LocalDate getActiveFrom() {
        return activeFrom;
    }

    public void setActiveFrom(LocalDate activeFrom) {
        this.activeFrom = activeFrom;
    }

    public LocalDate getActiveTo() {
        return activeTo;
    }

    public void setActiveTo(LocalDate activeTo) {
        this.activeTo = activeTo;
    }

    public UUID getId() {
        return id;
    }

    public void setId(UUID id) {
        this.id = id;
    }
}

File: _archive/archive_2024/kotlin/jpa-demo/src/main/kotlin/jpademo/Artist.kt
package jpademo

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import org.springframework.data.annotation.CreatedDate
import org.springframework.data.annotation.LastModifiedDate
import java.time.Instant
import java.util.UUID

@Entity
data class Artist (
    @Id
    var id: UUID = UUID.randomUUID(),
    var name: String = "",
    @CreatedDate
    @Column(name = "created_at")
    var created_at: Instant = Instant.now(),
    @LastModifiedDate
    @Column(name = "last_modified_at")
    var last_modified_at: Instant = Instant.now()
)